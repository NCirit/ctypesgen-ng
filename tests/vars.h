extern int errno;

extern struct {
	int a, b;
} pair;
