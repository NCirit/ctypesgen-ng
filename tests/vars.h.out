### This file automatically created by ctypesgen-ng, the commandline parameters passed to it are given below
### /app/ctypesgen.py --clang-flags "-Wno-error=implicit-int" --ignore-included --library test.so --headers tests/vars.h

import ctypes
import ctypes.util
import os


if not ('LIBRARY_PATH' in os.environ):
    if 'PYTHONPATH' in os.environ:
        os.environ['LIBRARY_PATH'] = os.environ['PYTHONPATH']
    else:
        os.environ['LIBRARY_PATH'] = os.path.dirname(os.path.abspath(__file__))

if os.name == "nt":
    if 'PYTHONPATH' in os.environ:
        os.environ['PATH'] += os.pathsep + os.environ['PYTHONPATH']
    else:
        os.environ['PATH'] += os.pathsep + os.path.dirname(os.path.abspath(__file__))
test_path = ctypes.util.find_library('test')
test = ctypes.CDLL(os.path.abspath(test_path))

var_errno = ctypes.c_int.in_dll(test, 'errno')


class __anon_struct1(ctypes.Structure):
    pass
__anon_struct1._pack_ = 1
__anon_struct1._fields_ = [
    ('a', ctypes.c_int),
    ('b', ctypes.c_int),
]

var_pair = __anon_struct1.in_dll(test, 'pair')
